ruller.go

```diff
@@ -298,8 +301,27 @@ func mergeMaps(rinfo *ruleInfo, sourceMap map[string]interface{}, destMapP *map[
 	}
 }
 
+func Middleware(next http.Handler) http.Handler {
+    return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {		
+		w.Header().Set("Access-Control-Allow-Origin", origins)
+		w.Header().Set("Access-Control-Allow-Methods", allow_methods)
+		w.Header().Set("Access-Control-Allow-Headers", allow_headers)
+
+		if (r.Method == "OPTIONS") {
+			//handle preflight in here
+			logrus.Debugf("OPTIONS Method")
+		  } else {
+			next.ServeHTTP(w, r)
+		  }
+    })
+}
+
+
 //StartServer Initialize and start REST server
 func StartServer() error {
+	origins = *(flag.String("origins", "", "Allowed Origins"))
+	allow_methods =  *(flag.String("allow-methods", "", "Allowed Methods"))
+	allow_headers = *(flag.String("allow-headers", "", "Allowed Headers"))
```

Comment:
Please use a default value here to easy usage. Do a research on common headers for being the default, pls.

Reply:
Nice!
So I will put the following values:

origins: ""
allow_methods: "POST, GET, OPTIONS"
allow_headers: "Accept, Accept-Encoding, Cache-Control, User-Agent, Accept-Language, Content-Type"

What about them as default? Do you agree? Or do you want to add more values?

Reply:
Yes! Great! Go ahead!

Reply:
Done!